### PR TITLE
Leverage readiness script

### PR DESCRIPTION
**Pull request by daxmc99**
Use script from https://github.com/sourcegraph/sourcegraph/pull/60365
### Checklist

- [x] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [x] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md)
- [ ] Update [Kubernetes update doc](https://docs.sourcegraph.com/admin/updates/kubernetes)

### Test plan

CI

### CODE REVIEW

**Review by keegancsmith — APPROVED** [review]
if you plan on backporting this, can you move the changelog entry into the 5.3.0 section?

Edit: nevermind on the backport, see here https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/419#issuecomment-1942062087

File: charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml [hunk 1]
```diff
@@ -65,15 +65,7 @@ spec:
...
-            - |
-              #!/bin/bash
-              response=$(
-                redis-cli ping
-              )
-              if [ "$response" != "PONG" ]; then
-                echo "$response"
-                exit 1
-              fi
+            - readiness.sh
```
Comment (eseliger) [inline charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml:68]: does this depend on the cwd? should we use `/readiness.sh` instead? Not a probe expert, so not sure how the WD is set

Comment (daxmc99) [reply charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml]: Updated! I think it uses the root but this is more explicit regardless

**Review by eseliger — APPROVED** [review]

**Comment by daxmc99** [comment]
Hmm, I'm struggling to find this new script in our images
https://github.com/sourcegraph/sourcegraph/pull/60365#issuecomment-1942088509

**Comment by daxmc99** [comment]
Works locally with this override.yaml file
```
# Disable SC creation
storageClass:
  create: false
  name: standard

# Disable resources requests/limits
sourcegraph:
  localDevMode: true
  image:
    defaultTag: "262142_2024-02-13_5.2-4db83b41b419"
    useGlobalTagAsDefault: true
# More values to be added in order to test your change

```

**Comment by daxmc99** [comment]
@sourcegraph/release I updated the values.yaml to `262142_2024-02-13_5.2-4db83b41b419` 
Otherwise this change will prevent the current chart from starting

**Pull request merged**